src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp

```diff
@@ -0,0 +1,149 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Types/Index.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace dynamics
+{
+
+using ostk::core::types::Index;
+
+using ostk::math::object::VectorXd;
+
+Tabulated::Tabulated(
+    const Array<Instant>& anInstantArray,
+    const MatrixXd& aContributionProfile,
+    const Array<Shared<const CoordinatesSubset>>& aWriteCoordinatesSubsets
+)
+    : Dynamics("Tabulated"),
+      contributionProfile_(aContributionProfile),
+      instants_(anInstantArray),
+      writeCoordinatesSubsets_(aWriteCoordinatesSubsets)
+{
+    if (anInstantArray.getSize() != (Index)aContributionProfile.rows())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of rows as the number of instants."
+        );
+    }
+
+    if (aWriteCoordinatesSubsets
+            .map<Index>(
+                [](const auto& coordinatesSubset)
+                {
+                    return coordinatesSubset->getSize();
+                }
+            )
+            .reduce(std::plus<Index>()) != (Index)aContributionProfile.cols())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Contribution profile must have the same number of columns as the sum of the sizes of the write "
+            "coordinates subsets."
+        );
+    }
+
+    VectorXd timestamps(anInstantArray.getSize());
```

Comment:
Yeah that's a good point, can use the first instant from the array.

Reply:
Also since we store all the instants, I'll pass on storing the startInstant_ and endInstant_ as it can be accessed via `instants_.accessFirst()` and `instants_.accessLast()`